core/app/models/spree/order.rb

```diff
@@ -735,8 +735,7 @@ def process_payments_before_complete
         true
       else
         saved_errors = errors[:base]
-        payment_failed!
-        saved_errors.each { |error| errors.add(:base, error) }
+        fail_transition(:payment_processing, saved_errors)
```

Comment:
Did you consider passing a block to the method instead? This way we can keep the logic here here.